### ISSUE TITLE
Bug in RemoveBins

### ISSUE THREAD

**Issue by mducle** [mantidproject/mantid#13612]
RemoveBins does not work if WorkspaceIndex is specified and does not equal 0. 

To reproduce:

```python
w=Load(Filename='MAR20371',OutputWorkspace='MAR20371')
    ...: w=RemoveBins(InputWorkspace="MAR20371", OutputWorkspace="MAR20371", XMin=0, XMax=2000, WorkspaceIndex=0)
    ...: x=w.readX(0); y=w.readY(0);
    ...: min([ x[i] for i in range(0,len(y)) if y[i]>0 ])
    ...: 
Out[1]: 2005.0

w=Load(Filename='MAR20371',OutputWorkspace='MAR20371')
    ...: w=RemoveBins(InputWorkspace="MAR20371", OutputWorkspace="MAR20371", XMin=0, XMax=2000, WorkspaceIndex=1)
    ...: x=w.readX(1); y=w.readY(1);
    ...: min([ x[i] for i in range(0,len(y)) if y[i]>0 ])
    ...: 
Out[2]: 5.0
```
The problem is in 147-150 of RemoveBins.cpp which contains the block that calculates the start and end indices, either if the units have changed or if it is the first histogram. However, this block is after an if statement which shortcuts the loop over the histograms (in line 138), so is never executed if Workspace!=0 is specified by the user. This leaves the default values start=end=0, so no bins are removed.

One fix is to remove the `if` statement (remove lines 147 and 150), so that block always execute, or to move the block to line 134, above the `continue` statement. 

Current code is:

```C++
135     // If just operating on a single spectrum and this isn't it, go to next
136     // iteration
137     if (singleSpectrum && (i != index))
138       continue;
139 
140     double startX(m_startX), endX(m_endX);
141     // Calculate the X limits for this spectrum, if necessary
142     if (unitChange) {
143       this->transformRangeUnit(i, startX, endX);
144     }
145 
146     // Calculate the bin indices corresponding to the X range, if necessary
147     if (recalcRange || !i) {
148       start = this->findIndex(startX, X);
149       end = this->findIndex(endX, X);
150     }
```

**Comment by NickDraper** [comment]
Duc,  Thanks for such a helpful bug report.

This should make it into the nightly builds from 20/11/2015, and be included in Release 3.6, due in Feb 2016

**Comment by mducle** [comment]
Thank you!
